### ISSUE TITLE
`-Wimplicit-int-float-conversion` or `-Wconversion` don't report "float -> int" conversion warnings

### ISSUE THREAD

**Issue by simco50** [microsoft/DirectXShaderCompiler#7142]
**Description**
I expect `-Wimplicit-int-float-conversion` or `-Wconversion` to produce a warning when doing implicit "float -> int" conversion.
Example in Clang: https://godbolt.org/z/WahdMPs71

**Steps to Reproduce**

```
StructuredBuffer<uint> In;
RWStructuredBuffer<float> Out;

[numthreads(1,1,1)]
void main(uint thread : SV_DispatchThreadID)
{
    Out[thread] = In[thread];
}
```

Commandline: `-Wconversion -WX -T cs_6_6 -E main`

Godbolt: https://godbolt.org/z/6xKasdxdx

**Actual Behavior**
DXC produces no warnings

**Environment**
- DXC version 1.8.2407
- Host Operating System W11 23H2
